server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -110,6 +135,1045 @@ void beforeEach() throws Exception {
+                    .build(), new ByteArrayInputStream(CONTENT)).block();
+
+                String query =
+                    "{" +
+                    "  \"combinator\": \"and\"," +
+                    "  \"criteria\": [" +
+                    "    {" +
+                    "      \"fieldName\": \"subject\"," +
+                    "      \"operator\": \"equalsIgnoreCase\"," +
+                    "      \"value\": \"SUBJECT Of the mail\"" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+                String taskId =
+                    given()
+                        .body(query)
+                    .when()
+                        .post(USER.asString())
+                        .jsonPath()
+                        .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                .when()
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", is("completed"));
+
+                assertThat(hasAnyMail(USER)).isFalse();
+            }
+        }
+
+        @Nested
+        class DeletionDateTest {
+
+            @Test
+            void restoreShouldAppendMessageToMailboxWhenMatchingDeletionDateBeforeOrEquals() throws Exception {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+
+                String query =
+                    "{" +
+                    "  \"combinator\": \"and\"," +
+                    "  \"criteria\": [" +
+                    "    {" +
+                    "      \"fieldName\": \"deletionDate\"," +
+                    "      \"operator\": \"beforeOrEquals\"," +
+                    "      \"value\": \"" + DELETION_DATE.plusHours(1).toString() + "\"" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+                String taskId =
+                    given()
+                        .body(query)
+                    .when()
+                        .post(USER.asString())
+                        .jsonPath()
+                        .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                .when()
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", is("completed"));
+
+                assertThat(restoreMessageContents(USER))
+                    .hasSize(1)
+                    .hasOnlyOneElementSatisfying(messageIS -> assertThat(messageIS).hasSameContentAs(new ByteArrayInputStream(CONTENT)));
+            }
+
+            @Test
+            void restoreShouldNotAppendMessageToMailboxWhenNotMatchingDeletionDateBeforeOrEquals() throws Exception {
+                vault.append(USER, DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+
+                String query =
+                    "{" +
+                    "  \"combinator\": \"and\"," +
```

Comment:
why do you use a composed criterion when you could express the same thing we a simple one?

Reply:
Like 

```
{
   "and" : [ ... ]
}
```

?????

Reply:
@mbaechler I think the top object inside JSON body should be the composed criterion. If I allow to pass a list of criterion as the top object, IDK how to make the DTO structure to satisfy JSON deserializations.

Reply:
You can imagine having `AndCriterion` + `OrCriterion` classes.

Then you look for `"and'` `"or"` fields and thus can choose the appropriate class.

In short: Let's agree on the API, we will always be able to easily implement it after....

In general: Let's not technical details affect our design too much.

Reply:
Sorry If my comment was hard to understand. What I was saying is : you have a single criterion here, so it doesn't make sense to wrap it into a "and", there's nothing to combine really.

Reply:
got it! fixup is at b833bcb

Reply:
not exactly. You should not have a list of criteria

Reply:
@mbaechler Can you provide an example of the exact JSON you expect?

Reply:
When there's a single criterion, we should just use : 
```json
{
"fieldName": "subject",
"operator": "contains",
"value": "subject contains"
}
```
We should use combinator only when we have something to combine